.github/workflows/ci.yml

```diff
@@ -1,39 +1,64 @@
-# Build gen-smtp on various OTP versions
-
 name: CI
-on:
-  push:
-    branches:
-      - master
-  pull_request:
-    branches:
-      - master
+
+on: [push, pull_request]
+
 jobs:
+
   ci:
-    name: Checks and tests on OTP-${{matrix.otp_vsn}}, ${{matrix.os}}, ${{matrix.profile}}
-    runs-on: ${{matrix.os}}
+    name: Test on OTP ${{ matrix.otp }} / Profile ${{ matrix.profile }}
+    runs-on: ${{ matrix.os }}
 
     strategy:
       matrix:
-        otp_vsn: [21.3, 22.3, 23.3, 24.1]
         os: [ubuntu-latest]
-        profile: [test, ranch18, ranch20]
-        exclude:
-            - otp_vsn: 24.1     # ranch 1.7 does not work on OTP24+
-              profile: test
-
-    container:
-      image: erlang:${{matrix.otp_vsn}}
+        otp: ["24", "23", "22"]
+        rebar3: ["3.17.0"]
+        profile: [test, ranch_v2]
+        include:
+          - os: ubuntu-latest
+            otp: "21"
+            rebar3: "3.15.2"
+            profile: "test"
+          - os: ubuntu-latest
+            otp: "21"
+            rebar3: "3.15.2"
+            profile: "ranch_v2"
 
     steps:
       - uses: actions/checkout@v2
-      - name: xref
+
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: ${{ matrix.otp }}
+          rebar3-version: ${{ matrix.rebar3 }}
+
+      - name: Cache Hex packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/rebar3/hex/hexpm/packages
+          key: ${{ runner.os }}-hex-${{ hashFiles('**/rebar.lock') }}
+          restore-keys: ${{ runner.os }}-hex-
+
+      - name: Cache Dialyzer PLTs
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/rebar3/rebar3_*.plt
+            _build/dialyzer/rebar3_*.plt
```

Comment:
is it really `.plt`? On my computer it creates `_plt`…

Reply:
No it's is indeed not. And I was thinking about why a second CI run isn't faster. Thanks!